drivers/i2c/i2c_ll_stm32.h

```diff
@@ -24,6 +24,7 @@ struct i2c_stm32_data {
 #ifdef CONFIG_I2C_STM32_INTERRUPT
 	struct k_sem device_sync_sem;
 #endif
+	struct k_sem		sem_lock;
```

Comment:
What about bus_lock instead of k_sem, if this is really what it is about?

Reply:
agree. I'll change the name!

Reply:
What about transfer_mutex?
Can also be bus_lock, but I would prefer to have the word mutex in it.

Reply:
bus_mutex?

Reply:
@erwango 
I just pushed a new version and, believe it or not, I came to the same name (bus_mutex) as well!